src/ompl/geometric/src/PathSimplifier.cpp

```diff
@@ -215,27 +215,29 @@ bool ompl::geometric::PathSimplifier::ropeShortcutPath(PathGeometric &path, doub
     // Store the cumulative costs
     // costs[i] contains the cumulative cost of the path up to and including state i
     std::vector<base::Cost> costs(states.size(), obj_->identityCost());
-    for (unsigned int i = 1; i < costs.size(); ++i)
+    for (std::size_t i = 1; i < costs.size(); ++i)
     {
         costs[i] = obj_->combineCosts(costs[i - 1], obj_->motionCost(states[i - 1], states[i]));
     }
 
     bool result = false;
+    ompl::base::Cost equivalenceCost(equivalenceTolerance * delta);
+    obj_->setCostThreshold(equivalenceCost);
```

Comment:
I like how you've done the equivalence cost now, but I don't think you should modify the optimization objective, as this could affect planner behavior outside of simplification.